src/github/github.installation.controller.ts

```diff
@@ -0,0 +1,203 @@
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Get('is-allowed/:organizationName')
+  async isOrganizationAllowed(
+    @Param('organizationName') organizationName: string,
+  ): Promise<boolean> {
+    try {
+      return this.githubInstallationService.isOrganizationAllowed(
+        organizationName,
+      );
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Get('is-user-allowed')
+  async isUserAllowed(
+    @Query('organizationName') organizationName: string,
+    @Query('githubUsername') githubUsername: string,
+  ): Promise<boolean> {
+    try {
+      return this.githubInstallationService.isUserAllowed(
+        organizationName,
+        githubUsername,
+      );
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Put('add-user/:organizationName')
+  async addAllowedUser(
+    @Param('organizationName') organizationName: string,
+    @Query('githubUsername') githubUsername: string,
+  ): Promise<AllowedOrganization> {
+    try {
+      return this.githubInstallationService.addAllowedUser(
+        organizationName,
+        githubUsername,
+      );
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Put('remove-user/:organizationName')
+  async removeAllowedUser(
+    @Param('organizationName') organizationName: string,
+    @Query('githubUsername') githubUsername: string,
+  ): Promise<AllowedOrganization> {
+    try {
+      return this.githubInstallationService.removeAllowedUser(
+        organizationName,
+        githubUsername,
+      );
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Put('update-status/:organizationName')
+  async updateStatus(
+    @Param('organizationName') organizationName: string,
+    @Query('status') newStatus: OrganizationStatus,
+  ): Promise<AllowedOrganization> {
+    try {
+      return this.githubInstallationService.updateStatus(
+        organizationName,
+        newStatus,
+      );
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
```

Comment:
PullProofAI

try {
    return await operation();
} catch (error) {
    this.handleException(error);
}
```

By centralizing the error handling logic, you promote code reusability and maintainability. This approach also ensures consistency across error handling in different controller methods.

This refactoring enhances readability, reduces code duplication, and makes it easier to update error handling logic in the future.


<!-- This is an auto-generated comment by PullProofAI -->